idx/cassandra/cassandra_test.go

```diff
@@ -285,3 +387,46 @@ func BenchmarkLoad(b *testing.B) {
 	ix.Init()
 	ix.Stop()
 }
+
+func BenchmarkIndexingWithUpdates(b *testing.B) {
+	cluster.Manager.SetPartitions([]int32{1})
+	keyspace = "metrictank"
+	hosts = "localhost:9042"
+	consistency = "one"
+	timeout = time.Second
+	numConns = 10
+	writeQueueSize = 10
+	protoVer = 4
+	updateInterval = time.Hour
+	updateCassIdx = true
+	ix := New()
+	tmpSession, err := ix.cluster.CreateSession()
+	if err != nil {
+		b.Skipf("can't connect to cassandra: %s", err)
+	}
+	tmpSession.Query("TRUNCATE metrictank.metric_idx").Exec()
+	tmpSession.Close()
+	if err != nil {
+		b.Skipf("can't connect to cassandra: %s", err)
+	}
+	ix.Init()
+	insertDefs(ix, b.N)
+
+	b.ReportAllocs()
+	b.ResetTimer()
+	var series string
+	var data *schema.MetricData
+	for n := 0; n < b.N; n++ {
+		series = "some.metric." + strconv.Itoa(n)
+		data = &schema.MetricData{
+			Name:     series,
+			Metric:   series,
+			Interval: 10,
+			OrgId:    1,
+			Time:     10,
+		}
+		data.SetId()
```

Comment:
now that an iteration is in the order of nanoseconds, the overhead of the Itoa call , instantiating data, and calling SetId is probably starting to get significant. maybe use StopTimer and StartTimer to take this stuff out of the equation.